habitat_sim/simulator.py

```diff
@@ -244,43 +288,124 @@ def initialize_agent(
                 )
 
         agent.set_state(initial_state, is_initial=True)
-        self._last_state = agent.state
+        if agent_id == len(self.__last_state):
+            self.__last_state.append(agent.state)
```

Comment:
Warning: mix of dictionaries and lists. What if agent_ids aren't sequential.

Reply:
@mathfac They have to be at init time currently. There is now way to create a non-sequential agent_id